### PR TITLE
Add Travis CI for supported targets

### PR DESCRIPTION
**Pull request by LDong-Arm**
This is based on https://github.com/ARMmbed/mbed-os-example-blinky/blob/development/.travis.yml but uses Mbed CLI 1.0 instead of CMake.

List of targets:
* ARM_MUSCA_S1
* ~ARM_MUSCA_S1_NS~ ([identical to ARM_MUSCA_S1](https://github.com/ARMmbed/mbed-os/pull/12955) for backward compatibility)
* ARM_MUSCA_B1
* ~ARM_MUSCA_B1_NS~ (identical to ARM_MUSCA_B1)
* K64F
* K66F
* NUCLEO_F429ZI
* FVP_MPS2_M0
* FVP_MPS2_M0P
* FVP_MPS2_M3
* FVP_MPS2_M4
* FVP_MPS2_M7
* GD32_F450ZI

Note: `CYTFM_064B0S2_4343W` is excluded for now due to Python failure of Cypress signing tool during Post-Build. This happens to any Mbed application for this target, I've created https://github.com/ARMmbed/mbed-os/issues/14022.

### CODE REVIEW

**Comment by LDong-Arm** [comment]
The list of items being built is here: https://github.com/ARMmbed/mbed-os-example-psa/pull/12/checks?check_run_id=1518492688
Once I've got some review, I can remove develop builds and only keep debug and release builds as @jainvikas8 suggested to me. Apart from that, do we want to build all these targets?

File: .travis.yml
Comment (jainvikas8) [inline .travis.yml:49]: Recommend to change to `mbed-psa-compile-test`

File: .travis.yml
Comment (jainvikas8) [inline .travis.yml:1]: Please change the commit message to have a description of the targets added and why?

Also please don't include any issues in it, unless that is addressed in that commit which is fixing it. By default, issues addressed will get closed

Comment (jainvikas8) [inline .travis.yml:1]: Please change the copyright year to only 2020

Comment (jainvikas8) [inline .travis.yml:1]: Please only mention profiles built.

File: .travis.yml
Comment (jainvikas8) [inline .travis.yml:151]: It would be good to use ccache support here - Please refer - https://github.com/ARMmbed/mbed-tools/blob/master/.travis.yml

**Review by jainvikas8 — CHANGES_REQUESTED** [review]

File: .travis.yml
Comment (LDong-Arm) [reply .travis.yml:49]: I'll change it. I used `mbed-compile-test` because all the other examples have "cmake-build-test" in Travis regardless of what they are building and was trying to follow the same: https://github.com/ARMmbed/mbed-os-example-blinky/blob/2a95fcd8dd9cdde303a3b8210afd3e93a0b9f09a/.travis.yml#L90
So `mbed-compile-test` meant the test for `mbed compile`

File: .travis.yml
Comment (LDong-Arm) [reply .travis.yml:1]: Good catch

Comment (LDong-Arm) [reply .travis.yml:1]: We previously had minimal commit messages and relied on PR description for details. The PR number can be retrieved from the merge commit. I'll elaborate the commit message.

**Review by LDong-Arm — COMMENTED** [review]
@jainvikas8 Thanks for the review, I'll fix them.

File: .travis.yml
Comment (LDong-Arm) [reply .travis.yml:1]: > https://github.com/ARMmbed/mbed-os-example-psa/pull/12#discussion_r540121071

But what if people ask why CYTFM_064B0S2_4343W is excluded? Is it kept track of?

File: .travis.yml
Comment (LDong-Arm) [reply .travis.yml]: Done

File: .travis.yml
Comment (LDong-Arm) [reply .travis.yml]: Updated commit message.

File: .travis.yml
Comment (LDong-Arm) [reply .travis.yml:170]: Waiting for CI to rerun to see if ccache is working (the stats is printed at the end of each compilation)

Comment (LDong-Arm) [reply .travis.yml:170]: Triggering a rebuild to see if there's any cache hits.

File: .travis.yml
Comment (jainvikas8) [reply .travis.yml]: Raised a issue - https://github.com/ARMmbed/mbed-os-example-psa/issues/14

File: .travis.yml
Comment (jainvikas8) [reply .travis.yml:170]: I'd suggest a specific cache name here

File: .travis.yml
Comment (LDong-Arm) [reply .travis.yml]: What I meant was, if we want a complete commit message, it should document known issues. I wasn't really asking that question.

**Comment by evedon** [comment]
> Once I've got some review, I can remove develop builds and only keep debug and release builds as @jainvikas8 suggested to me. 
I agree that we need release builds but I would use develop (default) instead of debug.

> Apart from that, do we want to build all these targets?
I will follow up on the list of targets and confirm. For now, let's have all supported targets.

**Comment by LDong-Arm** [comment]
> Apart from that, do we want to build all these targets?
> I will follow up on the list of targets and confirm. For now, let's have all supported targets.

Yea this is a concern. The CI run currently takes quite a long term.

File: .travis.yml
Comment (LDong-Arm) [reply .travis.yml:170]: Ah good point! I though CACHE_NAME was used by Mbed CLI v2 but it turns out to be a Travis feature. Will add it.

Comment (LDong-Arm) [reply .travis.yml]: Done

**Review by jainvikas8 — APPROVED** [review]
Ok to merge, if CI passes.

**Comment by LDong-Arm** [comment]
@evedon Missed it, updated

**Review by evedon — APPROVED** [review]

**Pull request merged**
